### PR TITLE
refactor: extract Python code to separate file

### PR DESCRIPTION
**Pull request by smjonas**
@quangnguyen30192 This is a replacement of #52. Can you test this branch and make sure nothing broke before I merge? Thank you.

### CODE REVIEW

File: python3/ultisnips_utils.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# using the UltiSnips API functions. Stores a list of dictionaries (one per
+# snippet) with the keys "trigger", "description", "options" and "value"
+# in the vim variable g:_cmpu_current_snippets.
+#
+# If 'expandable_only' is True, only expandable snippets are stored, otherwise
+# all snippets for the current filetype are added.
+
+
+def fetch_current_snippets(expandable_only):
+    if expandable_only:
```
Comment (quangnguyen30192) [inline python3/ultisnips_utils.py:13]: before_line is missing then the code doesn't work.  Let's add
```python
    before_line = vim_helper.buf.line_till_cursor
```

File: autoload/cmp_nvim_ultisnips.vim [hunk 2]
```diff
@@ -1,73 +1,3 @@
...
-before = vim_helper.buf.line_till_cursor
-visual_content = UltiSnips_Manager._visual_content
-expandable_only = vim.eval("a:expandable_only") == "True"
-if expandable_only:
-    snippets = UltiSnips_Manager._snips(before, True)
-else:
-    snippets = UltiSnips_Manager._snips("", True)
-
-for snippet in snippets:
-    is_context_snippet = snippet._context_code != None
```
Comment (quangnguyen30192) [inline autoload/cmp_nvim_ultisnips.vim:24]: Shall we still keep these in the new python file? I saw from line 24 - 32 is missed in the new python file

Comment (smjonas) [reply autoload/cmp_nvim_ultisnips.vim:24]: Thanks, I missed that

**Comment by quangnguyen30192** [comment]
Thanks, I've tested and it has no issue.

**Pull request merged**